src/js/components/modals/JobFormModal.js

```diff
@@ -54,6 +55,16 @@ const responseAttributePathToFieldIdMap = {
   '/labels': 'labels'
 };
 
+let cleanJSONspec = function (jsonDefinition) {
```

Comment:
Yes, and no. Both functions are using a different Config which we could pass as a parameter. I would prefer something like this if we want to have a more generic code:

```JS
function cleanJSONSpec(jsonSpec) {
  return cleanJSON(jsonSpec, JobConfig);
} 

function cleanJSONDefinition(jsonDefinition) {
  return cleanJSON(jsonDefinition, ServiceConfig);
}

function cleanJSON(json, config) {
  return Object.keys(json).filter(function (key) {
    return !config.BLACKLIST.includes(key);
  }).reduce(function (memo, key) {
    memo[key] = json[key];

    return memo;
  }, {});
}
```
This would remove code duplication but also remove the overhead of adding the config as a general parameter which would mean more code duplication.